agent/base_agent.py

```diff
@@ -93,3 +93,22 @@ def _initial_prompt(self, results: list[Result]) -> Prompt:
   @abstractmethod
   def execute(self, result_history: list[Result]) -> Result:
     """Executes the agent based on previous result."""
+
+
+if __name__ == "__main__":
+  # Make this a class method so that child agents can inherit and reuse?
+  import logging
+  import sys
+
+  import utils
+
+  agent_pickle = sys.argv[1]
+  result_history_pickle = sys.argv[2]
+  new_result_pickle = sys.argv[3]
```

Comment:
Is there a possibility we could avoid pickles?

Reply:
Unfortunately, I cannot think of a good solution yet because those variables are pretty complex.
What's the downside of using pickles?

Reply:
+1. Pickles are not recommended for security reasons. See the big red box in https://docs.python.org/3/library/pickle.html.

Reply:
Would signing the pickle file via `hmac` be safer than `json` (which may expose the data to humans)?

Reply:
I think we can just go with this for now to unblock this work, with a TODO look for a better replacement, or a disclaimer that there must never be untrusted data being passed in through this mechanism. 

>  (which may expose the data to humans)?

Can you elaborate on this concern?

Reply:
> (which may expose the data to humans)?

> Can you elaborate on this concern?

This is unimportant for now, but given we cannot trust the file transfer channel, I was thinking about:
1. Hashing and signing the file to ensure it is not tampered with.
2. Encrypting the file so that others cannot view its content (which may contain new vuls and reproducers).

Reply:
Using `dill` now : )